036-kraft-mode.md

```diff
@@ -0,0 +1,103 @@
+This is what gives the name to this proposal as well as to the feature gate discussed later.
+
+Strimzi aims to support the ZooKeeper-less Kafka and the KRaft protocol.
+Since the announcement of the KIP-500, we are making sure that Strimzi is as ready as possible for when the KRaft mode becomes production-ready.
+This includes for example adopting the new APIs based on Kafka instead of ZooKeeper in components such as User Operator or Cruise Control.
+But right now, you cannot really use Strimzi to run Kafka with the KRaft mode enabled.
+That makes it hard to continue working on some of the additional tasks needed to have Strimzi production-ready with KRaft in the future.
+For example the work on the readiness / liveness probes, Topic Operator improvements, upgrades etc.
+All of these tasks will be easier to work on with Strimzi supporting the KRaft mode out-of-the-box.
+
+## Proposal
+
+This proposal suggests to add provisional support for the KRaft mode.
+It will be disabled by default and will be protected by a new feature gate.
+When the feature gate is enabled, it will use the KRaft mode to deploy the Kafka cluster.
+It will simplify development and testing of new Strimzi features in the KRaft mode.
+And it would also allow to help with testing of the KRaft mode it self and raise any issues in the Apache Kafka project.
+
+The KRaft support proposed here is called provisional, because the API in the Kafka CR for configuring the Kafka nodes is not final and will be changed later.
+Most of the code implemented by this proposal is expected to be used for the final production-ready implementation.
+
+**The KRaft mode support suggested by this proposal will not be production-ready and should not be used outside of development.**
+
+### API changes
+
+In the initial implementation, there will be no changes to the `Kafka` custom resource and its API.
+When the KRaft mode is enabled, the operator will ignore any of the unsupported but required fields such as `.spec.zookeeper`.
+It will also validate the resource for any other unsupported features (such as authorization, JBOD storage etc.).
+
+The API is expected to change at a later phase before the KRaft support is considered production-ready.
+
+### KRaft deployment
+
+The initial implementation will support only a single type of KRaft deployment.
+All Kafka nodes will be created according to `.spec.kafka` section of the `Kafka` CR.
+It will respect the `.spec.kafka.replicas` field and deploy the corresponding number of Kafka nodes.
+All nodes will be assigned both the `controller` and `broker` KRaft roles.
+This deployment architecture is suitable for development and testing clusters.
+
+Other architectures, such as separate controller and broker nodes, will be not supported form the start.
+This is expected to change before the KRaft support is considered production-ready.
+
+### Feature Gate
+
+The new feature gate will be called `UseKraft`.
```

Comment:
Since pretty much everywhere else KRaft has a capital K and R should the feature gate be `UseKRaft`?

Reply:
Also is there docs anywhere that explain how to enable the feature gates? I checked the dev guide and couldn't see it anywhere

Reply:
> Also is there docs anywhere that explain how to enable the feature gates? I checked the dev guide and couldn't see it anywhere

General overview of enabling + we describe how to switch from the default state:
https://strimzi.io/docs/operators/in-development/configuring.html#ref-operator-cluster-feature-gates-str